placement/algo/helper.go

```diff
@@ -243,14 +264,14 @@ func (ph placementHelper) CanMoveShardToRack(shard uint32, from placement.HostSh
 	return true
 }
 
-func (ph placementHelper) canAssignHost(shard uint32, from, to placement.HostShards, rackCheck bool) bool {
+func (ph placementHelper) canAssignHost(shard uint32, from, to placement.HostShards) bool {
 	if to.ContainsShard(shard) {
 		return false
 	}
-	if rackCheck {
-		return ph.CanMoveShardToRack(shard, from, to.Host().Rack())
+	if ph.options.LooseRackCheck() {
```

Comment:
nit: can be `ph.options.LooseRackCheck() || ph.HasNoRackConflict(shard, from, to.Host().Rack())`